cpp/ql/lib/semmle/code/cpp/ir/implementation/raw/internal/TranslatedInitialization.qll

```diff
@@ -137,7 +137,10 @@ abstract class TranslatedInitialization extends TranslatedElement, TTranslatedIn
 
   final override string toString() { result = "init: " + expr.toString() }
 
-  final override Function getFunction() { result = expr.getEnclosingFunction() }
+  final override Declaration getFunction() {
+    result = expr.getEnclosingFunction() or
```

Comment:
Seems like we wind up with this exact same disjunction in many places. Maybe factor out a predicate like:
```ql
Declaration getEnclosingDeclaration(Expr e) {
  result = e.getEnclosingFunction() or result = e.getEnclosingVariable().(GlobalOrNamespaceVariable)
}
```
?